packages/tupaia-web/src/features/Dashboard/utils/useDashboard.ts

```diff
@@ -0,0 +1,44 @@
+// Utility hook to get the dashboard context
+
+import { useContext } from 'react';
+import { useParams } from 'react-router';
+import { useGAEffect } from '../../../utils';
+import { useDashboards } from '../../../api/queries';
+import { DashboardContext } from './DashboardContext';
+
+// Contains the dashboards, active dashboard, and export and subscribe state
+export const useDashboard = () => {
+  const { dashboardName, entityCode, projectCode } = useParams();
+  const { data: dashboards = [] } = useDashboards(projectCode, entityCode);
+  const {
+    selectedDashboardItems,
```

Comment:
I feel a little awkward about having `selectedDashboardItems` here. It's only used during whole dashboard exporting, and feels like it's not part of the core dashboard context.

I'd be happy for us to just keep it in local state in the ExportDashboard components like it used to be.